### PR TITLE
Added dependency to `swift-metrics` 

### PR DESCRIPTION
**Pull request by ptoffy**
Added a dependency to `swift-metrics` to be able to record metrics on queues.  

Metrics added:  
- Timer to see how long a job takes to complete 
- Counter to keep track of dispatched jobs 
- Two different counters to keep track of successes and failures

### CODE REVIEW

File: Sources/Queues/AsyncQueue.swift [hunk 1]
```diff
@@ -82,7 +83,8 @@ extension Queue {
         logger.trace("Pusing job to queue")
         try await self.push(id).get()
         logger.info("Dispatched job")
-        
+        Counter(label: "dispatched.jobs.counter", dimensions: [("queueName", self.queueName.string)]).increment()
```
Comment (0xTim) [inline Sources/Queues/AsyncQueue.swift:86]: Do we not want to record the lob name?

File: Sources/Queues/Queue.swift [hunk 2]
```diff
@@ -94,7 +96,12 @@ extension Queue {
             logger.trace("Pusing job to queue")
             return self.push(id)
         }.flatMapWithEventLoop { _, eventLoop in
-            logger.info("Dispatched job")
+            Counter(label: "dispatched.jobs.counter", dimensions: [("queueName", self.queueName.string)]).increment()
```
Comment (0xTim) [inline Sources/Queues/Queue.swift:99]: As above with the job name

File: Tests/QueuesTests/Utilities/CapturingMetricsSystem.swift [hunk 3]
```diff
@@ -0,0 +1,160 @@
+@testable import CoreMetrics
+import NIOConcurrencyHelpers
+import Vapor
+
+final class CapturingMetricsSystem: MetricsFactory {
```
Comment (0xTim) [inline Tests/QueuesTests/Utilities/CapturingMetricsSystem.swift:5]: Can we use https://github.com/apple/swift-metrics/blob/main/Sources/MetricsTestKit/TestMetrics.swift instead?

**Review by 0xTim — COMMENTED** [review]
Some points for discussion

File: Tests/QueuesTests/Utilities/TestMetrics.swift [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+
+/// Taken directly from `swift-cluster-memberships`'s own test target package, which
+/// adopts the `TestMetrics` from `swift-metrics`.
+///
+/// Metrics factory which allows inspecting recorded metrics programmatically.
+/// Only intended for tests of the Metrics API itself.
+///
+/// Created Handlers will store Metrics until they are explicitly destroyed.
+///
+public final class TestMetrics: MetricsFactory {
```
[the remark this answers is not in the document]
Comment (ptoffy) [reply Tests/QueuesTests/Utilities/TestMetrics.swift:40]: Ah sorry misunderstood that

**Review by 0xTim — APPROVED** [review]

**Pull request merged**